File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
//             for faster io operations
            ios_base::sync_with_stdio(false);
            cin.tie(NULL);
        
        unordered_map<string , vector<string>>map;
        int n = strs.size();
        string temp;
        for(int i = 0; i<n; i++){
            temp = strs[i];
            sort(strs[i].begin() , strs[i].end());
            map[strs[i]].push_back(temp);
        }
        vector<vector<string>> ans;
        for(auto itr = map.begin() ; itr != map.end() ; ++itr){
            ans.push_back(itr->second);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const map : Map<string , string[]> = new Map();
    for(const str of strs){
        const key = str.split('').sort().join('');
        if(map.has(key)){
            map.get(key).push(str);
        }else{
            map.set(key , [str]);
        }
    }
    return Array.from(map.values());
};